code/components/jomjol_flowcontroll/ClassFlowCNNGeneral.cpp

```diff
@@ -150,7 +156,10 @@ int ClassFlowCNNGeneral::PointerEvalHybridNew(float number, float number_of_pred
         // on first digit is no spezial logic for transition needed
         // we use the recognition as given. The result is the int value of the recognition
         // add precisition of 2 digits and round before trunc
-        result = (int) ((int) trunc(round((number+10 % 10)*100)) )  / 100;
+        // a number greater than 9.994999 is returned as 10, this leads to an error during the decimal shift because the NUMBERS[j]->ReturnRawValue is one digit longer.
+        // To avoid this, an additional test must be carried out, see lines 99 to 106
+        // Another alternative would be "result = (int) ((int) trunc(round((number+10 % 10)*1000))) / 1000;", which could, however, lead to other errors?
+        result = (int) ((int) trunc(round((number+10 % 10)*100)))  / 100;
```

Comment:
```suggestion
        result = (int) ((int) trunc(round((number+10 % 10)*100)) )  / 100;
```

Just to have no unnecessary code changes